### PR TITLE
Password change notifier

### PR DESCRIPTION
**Pull request by stgmsa**
# Description
Password change notifier for windows AD.

# Impacts
This will filter and report password change event happened within an AD

# Delete branch after merge
YES

# Checklist
- [yes] Document the feature
- [no] Add OpenAPI specification
- [no] Add unit tests
- [no] Add acceptance tests (TestLink)

# NEWS file entries
addons/AD/password_change_notifier.ps1

## New Features
* filter password change events and send this to packetfence using powershell.

## Enhancements
* filter password change events and send this to packetfence using powershell.

### CODE REVIEW

File: addons/AD/password_change_notifier.ps1 [hunk 1]
```diff
@@ -0,0 +1,36 @@
+##########################################################################################
+#Powershell script to unregister disabled Active Directory account based on the UserName.#
```
Comment (fdurand) [inline addons/AD/password_change_notifier.ps1]: description needs to be changed

File: addons/AD/password_change_notifier.ps1 [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        foreach ($entry in $xmlObject.Event.EventData.Data)
+        {
+            $e[$entry.Name] = $entry.InnerText
+        }
+    }
+    $eventArr += $e
+}
+
+$eventJson = $eventArr | ConvertTo-Json
+$url = "http://@PACKETFENCE_IP:@NTLMAUTH_LISTENING_PORT/password_change"
```
Comment (fdurand) [inline addons/AD/password_change_notifier.ps1]: the port of ntlm_auth_api is not exposed on the server (iptables prevent that)

Comment (stgmsa) [reply addons/AD/password_change_notifier.ps1]: updated. Please re-review.

File: addons/AD/password_change_notifier.ps1 [hunk 3]
```diff
@@ -0,0 +1,42 @@
+##########################################################################################
+# Powershell script to audit and notify a password change on an Active Directory.        #
+##########################################################################################
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+[Net.ServicePointManager]::ServerCertificateValidationCallback = { $true }
+
+$base_url = "https://#PACKETFENCE_IP:9999"
+$username = ""         # Username for the webservice, change to your own username.
```
Comment (fdurand) [inline addons/AD/password_change_notifier.ps1]: it's not the webservice but the api-frontend

File: addons/AD/password_change_notifier.ps1 [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+$password = ""         # Password for the webservice, change to your own password.
```
Comment (fdurand) [inline addons/AD/password_change_notifier.ps1]: it's not the webservice but the api-frontend

**Comment by fdurand** [comment]
What trigger the script ?

**Comment by stgmsa** [comment]
> What trigger the script ?

There's a configurable trigger on windows AD. 
Windows management -> Event viewer -> specific type of event -> launch a program.
then there will be an entry in "scheduled tasks" with a trigger based on events instead of time.
everytime a monitored type of event is logged, this script will be launched.

**Pull request merged**